ClientLibrary/PsiphonTunnel.go

```diff
@@ -220,12 +229,22 @@ func psiphon_tunnel_start(cConfigJSON, cEmbeddedServerEntryList, cClientPlatform
 
 	startTime := time.Now()
 
-	// Setup timeout signal
+	optionalTimeout := make(chan error)
+
+	if timeout != nil && *timeout != 0 {
+		// Setup timeout signal
 
-	runtimeTimeout := time.Duration(timeout) * time.Second
+		runtimeTimeout := time.Duration(*timeout) * time.Second
```

Comment:
`time.Duration` is an [`int64`](https://golang.org/src/time/time.go?s=21402:21421#L621), so there's a potential cast overflow here. I think that either the timeout param should be an `int64` (and a negative check) or there should be a check that the unsigned value is not greater than `math.MaxInt64`.

Reply:
Fixed in 9608aff